src/Stripe.net/Infrastructure/ParameterBuilder.cs

```diff
@@ -10,7 +10,7 @@ namespace Stripe.Infrastructure
 {
     internal static class ParameterBuilder
     {
-        public static string ApplyAllParameters(this StripeService service, object obj, string url, bool isListMethod = false)
+        public static string ApplyAllParameters(this StripeService service, StripeBaseOptions obj, string url, bool isListMethod = false)
```

Comment:
So I actually went with what I mentioned before: the method now expects a `StripeBaseOptions` instance instead of just an `object`. That way if we forget to derive from `StripeBaseOptions` the error will be immediately visible at compile time rather than at runtime. (I actually caught a few classes that you'd forgotten this way ;)

Plus, there's no need to muck with reflection to access `ExtraParams` now, so the code is a bit cleaner.